### PR TITLE
Add movingAverage

### PR DESCRIPTION
**Pull request by caisq**
Required by BatchNormalization layer training.

Another step toward fixing: https://github.com/tensorflow/tfjs/issues/20

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-core/963)

### CODE REVIEW

**Comment by nsthorat** [comment]
:lgtm_strong:

---

Review status: 0 of 3 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

*[src/ops/moving_average.ts, line 46 at r1](https://beta.reviewable.io/reviews/tensorflow/tfjs-core/963#-LAJwr2L44SbBluKefCh:-LAJwr2L44SbBluKefCi:b-e46nbf) ([raw file](https://github.com/tensorflow/tfjs-core/blob/fa59eb38d1f340f107e612e8b9fb014e3907bc88/src/ops/moving_average.ts#L46)):*
> ```XML
>    * in as `step`.
>    *
>    * @param v: The current moving average value.
> ```

no need for ":" here and below

---

*[src/ops/moving_average.ts, line 55 at r1](https://beta.reviewable.io/reviews/tensorflow/tfjs-core/963#-LAJvzA14byl-Y3ICoZg:-LAJvzA14byl-Y3ICoZh:bcltx2m) ([raw file](https://github.com/tensorflow/tfjs-core/blob/fa59eb38d1f340f107e612e8b9fb014e3907bc88/src/ops/moving_average.ts#L55)):*
> ```XML
>   @doc({heading: 'Operations', subheading: 'Moving Average'})
>   @operation
>   static movingAverage<R extends Rank, T extends Tensor<R>>(
> ```

You should be able to do this without `R extends Rank` and just `<T extends Tensor>`

---

*[src/ops/moving_average.ts, line 66 at r1](https://beta.reviewable.io/reviews/tensorflow/tfjs-core/963#-LAJwXvl40Duyg331rAK:-LAJwXvl40Duyg331rAL:b-2ed48d) ([raw file](https://github.com/tensorflow/tfjs-core/blob/fa59eb38d1f340f107e612e8b9fb014e3907bc88/src/ops/moving_average.ts#L66)):*
> ```XML
>     const oneMinusDecay = one.sub(decay);
> 
>     let update = (x.sub(v)).mul(oneMinusDecay);
> ```

you shouldnt need parens here, just do x.sub(b).mul(oneMinusDecay)

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/tensorflow/tfjs-core/963#-:-LAJwx9l4Sorx8SekXes:b-jbsf8z)*

**Comment by caisq** [comment]
Review status: 0 of 3 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[src/ops/moving_average.ts, line 46 at r1](https://beta.reviewable.io/reviews/tensorflow/tfjs-core/963#-LAJwr2L44SbBluKefCh:-LAPeaghD5lzPfai2W5E:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-core/blob/fa59eb38d1f340f107e612e8b9fb014e3907bc88/src/ops/moving_average.ts#L46)):*
<details><summary><i>Previously, nsthorat (Nikhil Thorat) wrote…</i></summary><blockquote>

no need for ":" here and below
</blockquote></details>

Done.

---

*[src/ops/moving_average.ts, line 55 at r1](https://beta.reviewable.io/reviews/tensorflow/tfjs-core/963#-LAJvzA14byl-Y3ICoZg:-LAPeg1M9Tp7cX65jzDn:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-core/blob/fa59eb38d1f340f107e612e8b9fb014e3907bc88/src/ops/moving_average.ts#L55)):*
<details><summary><i>Previously, nsthorat (Nikhil Thorat) wrote…</i></summary><blockquote>

You should be able to do this without `R extends Rank` and just `<T extends Tensor>`
</blockquote></details>

Done.

---

*[src/ops/moving_average.ts, line 66 at r1](https://beta.reviewable.io/reviews/tensorflow/tfjs-core/963#-LAJwXvl40Duyg331rAK:-LAPeipP3Kr4FLhDTeoP:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-core/blob/fa59eb38d1f340f107e612e8b9fb014e3907bc88/src/ops/moving_average.ts#L66)):*
<details><summary><i>Previously, nsthorat (Nikhil Thorat) wrote…</i></summary><blockquote>

you shouldnt need parens here, just do x.sub(b).mul(oneMinusDecay)
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/tensorflow/tfjs-core/963)*
